### PR TITLE
Added ability to record traceroutes for devices down due to ICMP

### PR DESCRIPTION
**Pull request by laf**
This can be used in alerting ($alert->debug['traceroute']) or viewed in the webui:

image

We could have the ability to turn on per device the ability to record this all the time.

At present it does it every time the device is down if the config option is on, we could do it when the status changes to down and no further.

Will add a small increase in poll time for the down device but that's probably not likely to be longer than a full poll for a device - but it could.

### CODE REVIEW

File: doc/Alerting/Templates.md [hunk 1]
```diff
@@ -50,6 +50,9 @@ Placeholders are special variables that if used within the template will be repl
 - ping min (if icmp enabled): `$alert->ping_min`
 - ping max (if icmp enabled): `$alert->ping_max`
 - ping avg (if icmp enabled): `$alert->ping_avg`
+- debug (array) This will contain:
```
Comment (murrant) [inline doc/Alerting/Templates.md]: Mention the config setting here too I think.

File: doc/Support/Configuration.md [hunk 2]
```diff
@@ -100,6 +100,15 @@ $config['icmp_check'] = false;
 
 If you would like to do this on a per device basis then you can do so under Device -> Edit -> Misc -> Disable ICMP Test? On
 
+#### traceroute
+LibreNMS uses traceroute / traceroute6 to record debug information when a device is down due to icmp AND you have `$config['debug']['run_trace] = true;`
```
Comment (swilsonskytap) [inline doc/Support/Configuration.md]: Missing single quote after run_trace.

Comment (murrant) [reply doc/Support/Configuration.md]: Can you send a pull request to fix?

Comment (swilsonskytap) [reply doc/Support/Configuration.md]: Sure.  I'm new to this, but I should be able to figure it out and get that going.

Comment (murrant) [reply doc/Support/Configuration.md]: Easy way go to the docs page and click the edit button ;)

Comment (swilsonskytap) [reply doc/Support/Configuration.md]: Thanks.  I'll get this figured out eventually :)   Anyway, looks like someone else corrected it.

**Comment by chasgames** [comment]
this is epic 👏🏻

**Comment by MrBlueSkye** [comment]
Great feature :100:

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/add-new-tool/7696/4

**Pull request merged**